apps/src/p5lab/spritelab/CoreLibrary.js

```diff
@@ -410,10 +471,7 @@ export default class CoreLibrary {
       return;
     }
     let name = opts.name;
-    let location = opts.location || {x: 200, y: 200};
-    if (typeof location === 'function') {
-      location = location();
```

Comment:
@mikeharv Do you have any idea why this is here/when the location will be a function and not just an Object with the shape `{x, y}`?

Reply:
Yes, but it's confusing that we are overloading the `location` variable like this. 

Lots of blocks call `addSprite()`. `opts.location` should be a function if it's defined. This could be a separate location block, or a field on the block, or neither. It might be clearer to write it like this:

```
const locationGenerator = opts.location || (() => ({x: 200, y: 200}));
const location = locationGenerator();
// ...
const sprite = this.p5.createSprite(location.x, location.y);
```

Reply:
When I use the pin block with the show/hide variable blocks, the location is an object. You're saying that's different for sprite locations? I'm torn on leaving the refactor as-is and converting the Sprite logic to what you have above, removing the `locationUtils` file entirely, and just duplicating it with the variable bubbles. Thoughts?

Reply:
I'm leaning towards leaving the `resolveLocation` the way it is because it works for everything, but 🤷‍♀️

Reply:
This solution works for me!

Apologies for missing your other question. Blocks that return a location do so either as a simple object value or as a function that returns the object value, e.g. `randomLocation`. In the case of the latter it's usually because we'd expect a potentially different position each time the code for a block is executed. I don't think it should be different for sprites and variable bubbles.